src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/PosixSignalRegistrationTests.Unix.cs

```diff
@@ -0,0 +1,241 @@
+        {
+            if (TriggersConsoleCancelKeyPress(signal))
+            {
+                // skip: xunit test runner returns non zero exit code.
+                return;
+            }
+
+            using SemaphoreSlim semaphore = new(0);
+            using var _ = PosixSignalRegistration.Create(signal, ctx =>
+            {
+                Assert.Equal(signal, ctx.Signal);
+
+                // Ensure signal doesn't cause the process to terminate.
+                ctx.Cancel = true;
+
+                semaphore.Release();
+            });
+            kill(signal);
+            bool entered = semaphore.Wait(Timeout);
+            Assert.True(entered);
+        }
+
+        [Fact]
+        public void SignalHandlerWorksForSecondRegistration()
+        {
+            PosixSignal signal = PosixSignal.SIGCONT;
+
+            for (int i = 0; i < 2; i++)
+            {
+                using SemaphoreSlim semaphore = new(0);
+                using var _ = PosixSignalRegistration.Create(signal, ctx =>
+                {
+                    Assert.Equal(signal, ctx.Signal);
+
+                    // Ensure signal doesn't cause the process to terminate.
+                    ctx.Cancel = true;
+
+                    semaphore.Release();
+                });
+                kill(signal);
+                bool entered = semaphore.Wait(Timeout);
+                Assert.True(entered);
+            }
+        }
+
+        [Fact]
+        public void SignalHandlerCalledNotCalledWhenDisposed()
+        {
+            PosixSignal signal = PosixSignal.SIGCONT;
+
+            using var registration = PosixSignalRegistration.Create(signal, ctx =>
+            {
+                Assert.False(true, "Signal handler was called.");
+            });
+            registration.Dispose();
+
+            kill(signal);
+            Thread.Sleep(100);
+        }
+
+        [Fact]
+        public void SignalHandlerCalledNotCalledWhenFinalized()
+        {
+            PosixSignal signal = PosixSignal.SIGCONT;
+
+            CreateDanglingRegistration();
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+
+            kill(signal);
+            Thread.Sleep(100);
+
+            [MethodImpl(MethodImplOptions.NoInlining)]
+            void CreateDanglingRegistration()
+            {
+                PosixSignalRegistration.Create(signal, ctx =>
+                {
+                    Assert.False(true, "Signal handler was called.");
+                });
+            }
+        }
+
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [InlineData(PosixSignal.SIGINT, true, 0)]
+        [InlineData(PosixSignal.SIGINT, false, 130)]
+        [InlineData(PosixSignal.SIGTERM, true, 0)]
+        [InlineData(PosixSignal.SIGTERM, false, 143)]
+        [InlineData(PosixSignal.SIGQUIT, true, 0)]
+        [InlineData(PosixSignal.SIGQUIT, false, 999)] // TODO: 131
```

Comment:
I've added an `expectedExitCode=999` here that should fail the test, but it doesn't.

```
<UnitTestResult executionId="8c96de44-f176-4cd8-886c-4f8b50dd7a6f" testId="3b74dd4e-1b7f-f062-5d9d-b682cd68ee40" testName="System.Tests.PosixSignalTests.SignalCanCancelTermination(signal: SIGQUIT, cancel: False, expectedExitCode: 999)" computerName="rex" duration="00:00:00.2979407" startTime="2021-06-22T14:10:34.9435192+02:00" endTime="2021-06-22T14:10:34.9435192+02:00" testType="13cdc9d9-ddb5-4fa4-a97d-d965ccfc6d4b" outcome="Passed" testListId="8c84fa94-04c1-424b-9868-57a2d4851a1d" relativeResultsDirectory="8c96de44-f176-4cd8-886c-4f8b50dd7a6f" />
```

@antonfirsov @stephentoub do you have an idea why the exit code isn't checked?

Reply:
ok, figured this out: when the delegate doesn't return an exit code, no exit code is checked.